### ISSUE TITLE
Number validation fails

### ISSUE THREAD

**Issue by ledniy** [Code-Forge-Net/remix-hook-form#52]
Hi there, seems like there's an issue with numbers validation on the backend side, coz if I have z.number() in the schema it will correctly pass validation on the frontend but the action will return an error:
```js
{
  message:  "Expected number, received string"
  type: "invalid_type"
}
```

I believe that's a bug as in readme you saying it should be parsed as a number
![image](https://github.com/Code-Forge-Net/remix-hook-form/assets/1562848/cdce0ad2-f4f7-4c99-ac98-eb05725e51e5)

But based on what I see in the `tryParseJSON` it will never parse it as a number, also in tests you expecting strings 🤷‍♂️

**Comment by AlemTuzlak** [comment]
@ledniy Yeah you're right, I've changed it to be passed to the backend as FormData instead of the JSON.stringify() method I was using before, this forces you to coerce types to make sure they are valid, eg z.number({ coerce: true }) for them to work, formData can be sent only as a string/file/blob type hence the reason for this. But in the docs it's outdated. Will change this

**Comment by AlemTuzlak** [comment]
fixed with 3.2
